### ISSUE TITLE
关于attention_mask 的疑问？

### ISSUE THREAD

**Issue by Bobo-y** [QwenLM/Qwen3-VL#1805]
对于自回归训练，为何没有看到前缀掩码 + 因果掩码的生成过程，输入的attention_mask 仅能区分真实和padding token, 我理解应该是

**Comment by JJJYmmm** [comment]
qwen-vl series do not use prefix mask.

**Comment by Bobo-y** [comment]
> qwen-vl series do not use prefix mask.

谢谢解答，还想确认一下，我目前在用的模型自回归训练时attention_mask 是如下这样构造的，所以按你的意思qwen-vl 系列训练时attention_mask 整个都是 下三角的吗？直接使用processor 输出的attention_mask 即可，不用特殊处理
                           img_token1 | img_token2 | prompt_token | answer_token1 |  answer_token2 |
img_token1          |           1           |        1          |         1             |           0            |           0             |

img_token2          |           1           |        1          |         1             |           0            |           0              |

prompt_token  |           1           |        1          |         1             |           0            |            0             |

answer_token1 |           1           |        1          |         1             |           1            |            0             |

answer_token2 |           1           |        1          |         1             |           1            |            1             |

**Comment by JJJYmmm** [comment]
是的，全下三角；你如果要改这种部分full mask，就得去改源码了，我记得有一个creat_causal_mask函数，你可以配合visual_pos_masks把visual token的full mask恢复出来，记得flash-attn不能用

**Comment by Bobo-y** [comment]
> 是的，全下三角；你如果要改这种部分full mask，就得去改源码了，我记得有一个creat_causal_mask函数，你可以配合visual_pos_masks把visual token的full mask恢复出来，记得flash-attn不能用

好的，谢谢 🤗
